frontend/src/metabase/nav/components/SearchResults.tsx

```diff
@@ -0,0 +1,113 @@
+import type { LegacyRef } from "react";
+import { useEffect, useMemo, useState } from "react";
+import { t } from "ttag";
+import { push } from "react-router-redux";
+import { useListKeyboardNavigation } from "metabase/hooks/use-list-keyboard-navigation";
+import { SearchResult } from "metabase/search/components/SearchResult";
+import { EmptyStateContainer } from "metabase/nav/components/SearchResults.styled";
+import EmptyState from "metabase/components/EmptyState";
+import type { SearchFilters } from "metabase/search/types";
+import {
+  DEFAULT_SEARCH_LIMIT,
+  SEARCH_DEBOUNCE_DURATION,
+} from "metabase/lib/constants";
+import { useSearchListQuery } from "metabase/common/hooks";
+import { useDispatch } from "metabase/lib/redux";
+import Search from "metabase/entities/search";
+import { Loader, Text, Stack } from "metabase/ui";
+import type { SearchModelType } from "metabase-types/api";
+import { useDebouncedEffect } from "metabase/hooks/use-debounced-effect";
+
+type SearchResultsProps = {
+  onEntitySelect?: (result: any) => void;
+  forceEntitySelect?: boolean;
+  searchText?: string;
+  searchFilters?: SearchFilters;
+  models?: SearchModelType[];
+};
+
+export const SearchResults = ({
+  onEntitySelect,
+  forceEntitySelect = false,
+  searchText,
+  searchFilters = {},
+  models,
+}: SearchResultsProps) => {
+  const dispatch = useDispatch();
+
+  const [debouncedSearchText, setDebouncedSearchText] = useState(searchText);
+  useDebouncedEffect(
```

Comment:
Would it be possible to use something from `react-use` instead?

Reply:
Yeah I can use `useDebounce`. Sorted